cmd/gateway-metrics.go

```diff
@@ -17,18 +17,16 @@
 package cmd
 
 import (
-	"sync"
-
 	"go.uber.org/atomic"
+	"net/http"
 )
 
 // Metrics - represents bytes served from backend
 // only implemented for S3 Gateway
 type Metrics struct {
 	BytesReceived atomic.Uint64
 	BytesSent     atomic.Uint64
-	RequestStats  map[string]int
-	sync.RWMutex
+	RequestStats  map[string]*atomic.Uint64
```

Comment:
If there is only 2 keys, maybe use a `[2]atomic.Uint64`?

Or if you need to preserve API: 
```suggestion
	RequestStats  struct {
		Get  atomic.Uint64 `json:"GET"`
		Head atomic.Uint64 `json:"HEAD"`
	}
```

Reply:
good point, let me update

Reply:
I checked and for Prometheus Labels to be added properly, it should be looped over like so https://github.com/minio/minio/blob/master/cmd/metrics.go#L235 so that way struct approach may not work.

Regarding `[2]atomic.Uint64`, it doesn't store the API which may be extended later, so IMHO current approach is fine..